test/spec/repeat.spec.js

```diff
@@ -70,6 +70,26 @@ describe( 'repeat functionality', () => {
             repeatButton.click();
             expect( [ ...form.view.html.querySelectorAll( '[name="/repdef/rep/num"]' ) ].map( i => i.value ) ).toEqual( [ '5', '5', '5' ] );
         } );
+
+        // https://github.com/enketo/enketo-core/issues/720
+        it( 'nested repeats are added correctly', () => {
+            const form = loadForm( 'nested-repeats.xml');
+            form.init();
+
+            // add repeats by clicking the add buttons
+            form.view.$.find( '.or-repeat-info[data-name="/data/repeat1/repeat11/repeat111"] .add-repeat-btn' ).click();
+            form.view.$.find( '.or-repeat-info[data-name="/data/repeat1/repeat11"] .add-repeat-btn' ).click();
+            form.view.$.find( '.or-repeat-info[data-name="/data/repeat1"] .add-repeat-btn' ).click();
```

Comment:
Gee,  I played around with this, and this bug is pretty hard to reproduce indeed!

I would use form.view.html.querySelector here, since you want to click only the first button. You do the same but it is less safe.